src/legacy/core_plugins/console/np_ready/public/lib/ace_token_provider/token_provider.test.ts

```diff
@@ -0,0 +1,190 @@
+
+import { AceTokensProvider } from '.';
+import { Token, TokensProvider } from '../../interfaces';
+
+describe('Ace (legacy) token provider', () => {
+  let aceEditor: Editor & { $el: any; autocomplete: any; update: any };
+  let tokenProvider: TokensProvider;
+  beforeEach(() => {
+    // Set up our document body
+    document.body.innerHTML = `<div>
+        <div id="ConAppEditor" />
+        <div id="ConAppEditorActions" />
+        <div id="ConCopyAsCurl" />
+      </div>`;
+
+    aceEditor = initializeInput(
+      $('#ConAppEditor'),
+      $('#ConAppEditorActions'),
+      {},
+      { applyCurrentSettings: () => {} },
+      null
+    );
+
+    aceEditor = initializeInput(
+      $('#ConAppEditor'),
+      $('#ConAppEditorActions'),
+      {},
+      { applyCurrentSettings: () => {} },
+      null
+    );
+    aceEditor.$el.show();
+    aceEditor.autocomplete._test.removeChangeListener();
+    tokenProvider = new AceTokensProvider(aceEditor.session);
+  });
+  afterEach(done => {
+    aceEditor.$el.hide();
+    aceEditor.autocomplete._test.addChangeListener();
+    aceEditor.update('', done);
+  });
+
+  const runTest = ({
+    input,
+    expectedTokens,
+    cb,
+    lineNumber = 1,
+  }: {
+    input: string;
+    expectedTokens: Token[] | null;
+    cb: () => void;
+    lineNumber?: number;
+  }) => {
+    aceEditor.update(input, function() {
+      const tokens = tokenProvider.getTokens(lineNumber);
+      expect(tokens).toEqual(expectedTokens);
+      cb();
+    });
+  };
+
+  describe('#getTokens', () => {
+    describe('base cases', () => {
+      test('case 1', function(cb) {
+        runTest({
+          input: `GET http://somehost/_search`,
+          expectedTokens: [
+            { type: 'method', value: 'GET', position: { lineNumber: 1, column: 1 } },
+            { type: 'whitespace', value: ' ', position: { lineNumber: 1, column: 4 } },
+            {
+              type: 'url.protocol_host',
+              value: 'http://somehost',
+              position: { lineNumber: 1, column: 5 },
+            },
+            { type: 'url.slash', value: '/', position: { lineNumber: 1, column: 20 } },
+            { type: 'url.part', value: '_search', position: { lineNumber: 1, column: 21 } },
+          ],
+          cb,
+        });
+      });
+
+      test('case 2', function(cb) {
+        runTest({
+          input: `GET http://test:user@somehost/`,
+          expectedTokens: [
+            { type: 'method', value: 'GET', position: { lineNumber: 1, column: 1 } },
+            { type: 'whitespace', value: ' ', position: { lineNumber: 1, column: 4 } },
+            {
+              type: 'url.protocol_host',
+              value: 'http://test:user@somehost',
+              position: { lineNumber: 1, column: 5 },
+            },
+            { type: 'url.slash', value: '/', position: { lineNumber: 1, column: 30 } },
+          ],
+          cb,
+        });
+      });
+
+      test('case 3', function(cb) {
```

Comment:
It might be good to describe your assertions in these tests with more detail. For example, I'm not sure if the extra lines on lines 125-126 are intentional or accidental. I assume they're intentional, and we're verifying that empty lines are treated as terminating tokens (not sure what the right term is so I'm making this one up) by the token provider? And if so then is this a duplicate of `case 2 - empty lines`  on line 173?

Reply:
:+1: